Home_work24/Vector2d.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+#include <math.h>
+#include<iostream>
+namespace VectorRelativeState {
+	enum {
+		Identical, coDirected, OppositeDirected, AcuteAngle, ObtuseAngle, RightAngle
+	};
+}
+class Vector2D {
+private:
+	float x = 0.0;
+	float y = 0.0;
+	enum class VectorRelativeState {
+		Identical, coDirected, OppositeDirected, AcuteAngle, ObtuseAngle, RightAngle
+	};
+	
+public:
+
+	Vector2D(float x, float y) : x{ x }, y{ y } {}
+	Vector2D(float x0, float y0, float x1, float y1) : x{ x1 - x0 }, y{ y1 - y0 } {}
+	Vector2D(const Vector2D&) = default;
+	float dotProduct( const Vector2D& other) const {
+		return x*other.x + y*other.y;
+	}
+	Vector2D negative() const {
+		return { x * (-1), y * (-1) };
+	}
+
+	Vector2D operator=(const Vector2D& secondVector) const {
+		return { secondVector.x, secondVector.y };
+	}
+	Vector2D operator+(const Vector2D& secondVector) const  {
+		return { x + secondVector.x, y + secondVector.y };
+	}
+	Vector2D operator-(const Vector2D& secondVector) const {
+		return { x - secondVector.x, y - secondVector.y };
+	}
+
+	friend Vector2D operator+(const Vector2D& firstVector, const Vector2D& secondVector) {
+		return { firstVector.x + secondVector.x, firstVector.y + secondVector.y };
+	}
+	friend Vector2D operator-(const Vector2D& firstVector, const Vector2D& secondVector) {
+		return { firstVector.x - secondVector.x, firstVector.y - secondVector.y };
+	}
+	float operator()() const {
+
+		return module();
+	}
+	float& operator[](int i) {
+		if (i == 0) return x;
+		else if (i == 1) return y;
```

Comment:
`not all control paths return values`
That's kind of message you should have received with this code.
What if i == 2?